crates/evm/src/evm/executor.rs

```diff
@@ -37,6 +39,21 @@ where
         *self.evm.tx_mut() = create_tx_env(tx);
         self.evm.transact_commit()
     }
+
+    /// Runs a single transaction in the configured environment and proceeds
+    /// to return the result and state diff (without applying it).
+    fn transact(
+        &mut self,
+        tx: &TransactionSignedEcRecovered,
+    ) -> Result<ResultAndState, EVMError<Infallible>> {
+        self.evm.context.external.set_current_tx_hash(tx.hash());
+        *self.evm.tx_mut() = create_tx_env(tx);
+        self.evm.transact()
+    }
+
+    fn commit(&mut self, state: State) {
```

Comment:
add comment to function

Reply:
Well there is a possibility to commit a state that does not belong to the current evm/block.

Imagine:

L2 block №42:

```
let state = evm.transact(..);
// save state to heap
```

L3 block №43:

```
// load state from heap
let state = ...;
evm.commit();
```

Is it okay? If no, State must be bound to an evm via lifetime.